test/test_from_sdist.py

```diff
@@ -0,0 +1,214 @@
+
+    # make an sdist of the project
+    sdist_dir = tmp_path / "sdist"
+    sdist_dir.mkdir()
+    sdist_path = make_sdist(project, sdist_dir)
+
+    # build the wheels from sdist
+    actual_wheels = cibuildwheel_from_sdist_run(
+        sdist_path,
+        add_env={"CIBW_BUILD": "cp39-*"},
+    )
+
+    # check that the expected wheels are produced
+    expected_wheels = [w for w in utils.expected_wheels("spam", "0.1.0") if "cp39" in w]
+    assert set(actual_wheels) == set(expected_wheels)
+
+    # check that before-build was run
+    captured = capfd.readouterr()
+    assert "test log statement from before-build 8419" in captured.out
+
+
+def test_internal_config_file_argument(tmp_path, capfd):
+    # make a project with a config file inside
+    project = test_projects.new_c_project(
+        setup_cfg_add="include_package_data = True",
+    )
+    project.files["wheel_build_config.toml"] = textwrap.dedent(
+        """
+        [tool.cibuildwheel]
+        before-all = 'echo "test log statement from before-all 1829"'
+        """
+    )
+    project.files["MANIFEST.in"] = textwrap.dedent(
+        """
+        include wheel_build_config.toml
+        """
+    )
+
+    # make an sdist of the project
+    sdist_dir = tmp_path / "sdist"
+    sdist_dir.mkdir()
+    sdist_path = make_sdist(project, sdist_dir)
+
+    # build the wheels from sdist, referencing the config file inside
+    actual_wheels = cibuildwheel_from_sdist_run(
+        sdist_path,
+        add_env={"CIBW_BUILD": "cp39-*"},
+        config_file="{package}/wheel_build_config.toml",
+    )
+
+    # check that the expected wheels are produced
+    expected_wheels = [w for w in utils.expected_wheels("spam", "0.1.0") if "cp39" in w]
+    assert set(actual_wheels) == set(expected_wheels)
+
+    # check that before-all was run
+    captured = capfd.readouterr()
+    assert "test log statement from before-all 1829" in captured.out
+
+
+def test_argument_passthrough(tmp_path, capfd):
+    basic_project = test_projects.new_c_project()
+
+    # make an sdist of a project
+    sdist_dir = tmp_path / "sdist"
+    sdist_dir.mkdir()
+    sdist_path = make_sdist(basic_project, sdist_dir)
+
+    # make a call that should pass some args through to cibuildwheel
+    # this asks cibuildwheel to print the ppc64le build identifiers
+    process = subprocess.run(
+        [
+            sys.executable,
+            "-m",
+            "cibuildwheel",
+            str(sdist_path),
+            "--platform",
+            "linux",
+            "--archs",
+            "ppc64le",
+            "--print-build-identifiers",
+        ],
+        env={
+            **os.environ,
+            "CIBW_BUILD": "cp38-*",
+        },
+        check=True,
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+    )
+
+    # fmt: off
+    assert process.stdout == textwrap.dedent(
+        """
+        cp38-manylinux_ppc64le
+        cp38-musllinux_ppc64le
+        """
+    ).lstrip()
+    # fmt: on
```

Comment:
I think we can remove this test, now? It doesn't test anything now that the calling method is the same

```suggestion
```